### PR TITLE
Update user profile and user stories in Developer Guide

### PR DESCRIPTION
**Pull request by zacchaeuschok**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -257,33 +257,147 @@ _{Explain here how the data archiving feature will be implemented}_
...
+    * 3a1. Condonery shows an error message.
+
+      Use case resumes at step 2.
+
+*{More to be added}*
+
+### Non-Functional Requirements
+
+1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
+2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
```
Comment (yuehernkang) [inline docs/DeveloperGuide.md:384]:
```suggestion
2.  Should be able to hold up to 1000 listings and contacts without a noticeable sluggishness in performance for typical usage.
```

**Review by lkwlkww — APPROVED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (zacchaeuschok) [reply docs/DeveloperGuide.md]: we can add this change under a different pull request for NFRs

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -257,33 +257,147 @@ _{Explain here how the data archiving feature will be implemented}_
...
+* [Configuration guide](Configuration.md)
+* [DevOps guide](DevOps.md)
+
+--------------------------------------------------------------------------------------------------------------------
+
+## **Appendix: Requirements**
+
+### Product scope
+
+**Target user profile**:
```
Comment (lkwlkww) [inline docs/DeveloperGuide.md]: do you think we should include that the user should be a condo seller, etc.?

**Review by misterpuffin — APPROVED** [review]
lgtm!

**Pull request merged**